### DISCUSSION TITLE
Storybook 7 with Angular 15 and browserTarget

### DISCUSSION THREAD

**Discussion by m4tuna** [storybookjs/storybook#21462] [Help]
What is the right way to configure my angularBrowserTarget for Storybook 7 to work properly with Angular 15 and Typescript? 

Here's my setup:

  System:
    OS: macOS 13.1
    CPU: (8) arm64 Apple M1
  Binaries:
    Node: 16.15.0 - ~/.nvm/versions/node/v16.15.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.15.0/bin/yarn
    npm: 8.5.5 - ~/.nvm/versions/node/v16.15.0/bin/npm
  Browsers:
    Chrome: 105.0.5195.125
    Firefox: 109.0.1
    Safari: 16.2
  npmPackages:
    @storybook/addon-actions: ~7.0.0-beta.60 => 7.0.0-beta.60
    @storybook/addon-essentials: ~7.0.0-beta.60 => 7.0.0-beta.60
    @storybook/addon-links: ~7.0.0-beta.60 => 7.0.0-beta.60
    @storybook/addons: ~7.0.0-beta.60 => 7.0.0-beta.60
    @storybook/angular: ~7.0.0-beta.60 => 7.0.0-beta.60
    @storybook/theming: ~7.0.0-beta.60 => 7.0.0-beta.60

I've seen a few posts and mention of this issue cropping up with Angular 13 that provided some insight, but I can't seem to shake the setup error:

ERR!
ERR! Error: angularBrowserTarget is undefined.
ERR!     at null.checkForLegacyBuildOptions (/Users/m42/git/halo/hdf/ui-components/node_modules/@storybook/angular/dist/server/framework-preset-angular-cli.js:84:11)
ERR!     at Object.webpackFinal (/Users/m42/git/halo/hdf/ui-components/node_modules/@storybook/angular/dist/server/framework-preset-angular-cli.js:16:5)
ERR!     at /Users/m42/git/halo/hdf/ui-components/node_modules/@storybook/core-common/dist/index.js:15:1046
ERR!  /Users/m42/git/halo/hdf/ui-components/node_modules/@storybook/angular/dist/server/framework-preset-angular-cli.js:84
ERR!     throw Error('angularBrowserTarget is undefined.');

Have tried in `main.ts`:

  angularOptions: {
    // Specify the target name here
    angularBrowserTarget: 'halo-ui-components:storybook',
  },

As well as updating my `angular.json` a few times. Any insight would be helpful. 

Thanks in advance, you fine Storybook folks!

**Comment by m4tuna** [comment]
bump

**Comment by bzh2610** [comment]
Can you share your angular.json config ? 

Storybook requires a browserTarget parameter like for your server configuration (usually named serve)

```JSON
"my-app": {
[...]
      "projectType": "application",
      "architect": {
[...]
        "serve-storybook": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "tsConfig": ".storybook/tsconfig.json",
            "assets": [  ],
            "index": "src/index.html",
            "main": "src/main.ts",
            "outputPath": "../app/src/main/webapp/",
            "styles": [],
          }
        },
        "storybook": {
          "builder": "@storybook/angular:start-storybook",
          "options": {
            "configDir": ".storybook",
            "browserTarget": "my-app:serve-storybook",
            "compodoc": false,
            "port": 6006,
            "styles": []
          }
        }
}
```

Also, make sure that Angular framework is configured in .storybook/main.js 

```javascript
  framework: {
    name: "@storybook/angular",
    options: {},
  },
```

**Reply by 87Dev** [reply]
Hello @bzh2610, 

Could you go through the entire configuration file to check it? I'm in the same situation and I can't get it to start.

Greetings

**Reply by bzh2610** [reply]
@87Dev The contents will vary according to your project, but you should have an "architect" configuration in which you can specify the options above. Edit "my-app" according to your app name, and make sure the directories I gave as example match your configuration. 

> I can't get it to start.

What logs do you get / which command did you run ?

Also, Angular docs may help: [https://angular.io/guide/workspace-config](https://angular.io/guide/workspace-config)

**Reply by 87Dev** [reply]
@bzh2610 Thanks for the help.

A few minutes ago I managed to get it to boot, I didn't read the documentation well.

Thanks for your time

**Reply by GuanyiX** [reply]
May I ask how did you get it worked? I faced the same issue @87Dev

**Reply by 87Dev** [reply]
@GuanyiX I solve the problem with this config

`
"projects": {
    "your-project": {
        "projectType": "library",
        "root": "projects/your-project",
        "sourceRoot": "projects/your-project/src",
        "schematics": {
            "@schematics/angular:component": {
                "style": "scss"
            }
        },
        "architect": {
            "build": {
                "builder": "@angular-devkit/build-angular:ng-packagr",
                "options": {
                    "project": "projects/your-project/ng-package.json"
                },
                "configurations": {
                    "production": {
                        "tsConfig": "projects/your-project/tsconfig.lib.prod.json"
                    },
                    "development": {
                        "tsConfig": "projects/your-project/tsconfig.lib.json"
                    }
                },
                "defaultConfiguration": "production"
            },
            "test": {
                "builder": "@angular-devkit/build-angular:karma",
                "options": {
                    "main": "projects/your-project/src/test.ts",
                    "tsConfig": "projects/your-project/tsconfig.spec.json",
                    "karmaConfig": "projects/your-project/karma.conf.js"
                }
            },
            "lint": {
                "builder": "@angular-eslint/builder:lint",
                "options": {
                    "cache": true,
                    "lintFilePatterns": ["projects/**/*.ts", "projects/**/*.html", "src/**/*.ts", "src/**/*.html"]
                }
            },
            "serve-storybook": {
                "builder": "@angular-devkit/build-angular:browser",
                "options": {
                    "tsConfig": ".storybook/tsconfig.json",
                    "assets": [],
                    "index": "src/index.html",
                    "main": "src/main.ts",
                    "outputPath": "../app/src/main/webapp/"
                }
            },
            "build-storybook": {
                "builder": "@storybook/angular:build-storybook",
                "options": {
                    "browserTarget": "your-project:build"
                }
            },
            "storybook": {
                "builder": "@storybook/angular:start-storybook",
                "options": {
                    "configDir": ".storybook",
                    "browserTarget": "your-project:build",
                    "compodoc": false,
                    "port": 6006,
                    "styles": ["./projects/your-project/styles/styles.scss", ".storybook/preview.scss"]
                }
            }
        }
    },
}`

Sorry the formating xD

**Reply by GuanyiX** [reply]
Thank you for you reply! @87Dev

Unfortunately I'm still not able to get it resolved. What I have done is 

1. Set up a brand new Angular v15 app
2. Run npx storybook@latest init to set up Storybook

After doing that, I can run npm run storybook and everything is working fine. However, when I changed the 'storybook' script in package.json to 'storybook dev -p 6006', and run it again, it throws this error


info => Starting manager..
info Addon-docs: using MDX2
info => Using implicit CSS loaders
ERR! Your Storybook startup uses a solution that is not supported.
ERR! You must use angular builder to have an explicit configuration on the project used in angular.json
ERR! Read more at:
ERR! - https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#sb-angular-builder)
ERR! - https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#angular13)
ERR! Error: angularBrowserTarget is undefined.
ERR!     at checkForLegacyBuildOptions (/Users/testuser/Desktop/test-storybook/node_modules/@storybook/angular/dist/server/framework-preset-angular-cli.js:84:11)
ERR!     at Object.webpackFinal (/Users/testuser/Desktop/test-storybook/node_modules/@storybook/angular/dist/server/framework-preset-angular-cli.js:16:5)
ERR!     at /Users/ testuser/Desktop/test-storybook/node_modules/@storybook/core-common/dist/index.js:15:1046
ERR!  Error: angularBrowserTarget is undefined.
ERR!     at checkForLegacyBuildOptions (/Users/testuser/Desktop/test-storybook/node_modules/@storybook/angular/dist/server/framework-preset-angular-cli.js:84:11)
ERR!     at Object.webpackFinal (/Users/testuser/Desktop/test-storybook/node_modules/@storybook/angular/dist/server/framework-preset-angular-cli.js:16:5)
ERR!     at /Users/testuser/Desktop/test-storybook/node_modules/@storybook/core-common/dist/index.js:15:1046

WARN Broken build, fix the error above.
WARN You may need to refresh the browser.

I'm using MacOS. 
My environment is:
npm: 8.19.2
node: v18.10.0
Angular: 15
Storybook: 7.0.18

Please leave any comments, thanks!

**Reply by GuanyiX** [reply]
I also tested with React, and found the default 'storybook' script is 'storybook dev -p 6006'. Does it mean Storybook CLI only works with React?

**Reply by 87Dev** [reply]
@GuanyiX your package command should be something like this

` "storybook": "ng run project-name:storybook",`

**Reply by GuanyiX** [reply]
Then I guess it could be a bug right? 

As per documentation https://www.npmjs.com/package/@storybook/angular, if you take a look at the FAQ section, my understanding is it should allow to have `"storybook": "storybook dev -p 6006"`  in Angular.

@87Dev
